tests/jobs/ahm_analysis/test_ahm_analysis.py

```diff
@@ -0,0 +1,421 @@
+    ks_matrix = pd.DataFrame()
+    dkeys = ["param1", "param2"]
+    ks_matrix = pd.DataFrame(sorted(dkeys), columns=["Parameters"])
+    np.random.seed(12345678)
+    prior_data = pd.DataFrame(
+        {
+            "param1": stats.norm.rvs(size=100, loc=0, scale=1),
+            "param2": stats.norm.rvs(size=100, loc=0.01, scale=1),
+        }
+    )
+    target_data = pd.DataFrame(
+        {
+            "param1": stats.norm.rvs(size=200, loc=0.5, scale=1.5),
+            "param2": stats.norm.rvs(size=200, loc=0.1, scale=1),
+        }
+    )
+    ks_matrix["WOPT:W1"] = ks_matrix["Parameters"].map(
+        ahmanalysis.calc_kolmogorov_smirnov(dkeys, prior_data, target_data)
+    )
+    ks_matrix.set_index("Parameters", inplace=True)
+    assert "param2" in ks_matrix.index
+    assert ks_matrix.loc["param1", "WOPT:W1"] == 0.18
+    assert ks_matrix["WOPT:W1"].max() <= 1
+    assert ks_matrix["WOPT:W1"].min() >= 0
+
+
+def test_check_names():
+    """test function check names to be used for prior and update case"""
+    ert_currentname = "default"
+    prior_name = None
+    target_name = "<ANALYSIS_CASE_NAME>"
+    p_name, tname = ahmanalysis.check_names(ert_currentname, prior_name, target_name)
+    assert p_name == ert_currentname
+    assert tname != "<ANALYSIS_CASE_NAME>"
+
+
+@pytest.mark.parametrize(
+    "misfit_data, prior_data, expected_msg",
+    [
+        (
+            pd.DataFrame(),
+            pd.DataFrame({"KEY:OP1": [0, 1, 2]}),
+            "Empty prior ensemble",
+        ),
+        (
+            pd.DataFrame(
+                {
+                    "MISFIT:KEY": [6],
+                }
+            ),
+            pd.DataFrame(),
+            "Empty parameters set for History Matching",
+        ),
+    ],
+)
+def test_raise_if_empty(misfit_data, prior_data, expected_msg):
+    """test function check that run fails if empty misfit or prior data"""
+    with pytest.raises(ValidationError, match=expected_msg):
+        ahmanalysis.raise_if_empty(
+            dataframes=[prior_data, misfit_data],
+            messages=[
+                "Empty prior ensemble",
+                "Empty parameters set for History Matching",
+            ],
+        )
+
+
+@pytest.mark.parametrize(
+    "input_map, expected_keys",
+    [
+        (
+            {"data_key_1": ["obs_1", "obs_2", "obs_3"]},
+            ["data_key_1"],
+        ),
+        (
+            {"data_key_1": ["obs_1", "obs_2", "obs_3"], "data_key_2": ["obs_4"]},
+            ["data_key_1", "data_key_2", "All_obs"],
+        ),
+        (
+            {
+                "data_key_1": ["obs_1", "obs_2", "obs_3"],
+                "data_key_2": ["obs_4"],
+                "data_key_3": ["obs_6"],
+            },
+            [
+                "data_key_1",
+                "data_key_2",
+                "data_key_3",
+                "All_obs-data_key_3",
+                "All_obs-data_key_2",
+                "All_obs-data_key_1",
+                "All_obs",
```

Comment:
Before adding the suggestion on implementation I exhausted this expected by including the values as well, I would argue that is a good thing to better test and document what it does:

```
(
            {"data_key_1": ["obs_1", "obs_2", "obs_3"]},
            {"data_key_1": ["obs_1", "obs_2", "obs_3"]},
        ),
        (
            {"data_key_1": ["obs_1", "obs_2", "obs_3"], "data_key_2": ["obs_4"]},
            {
                "data_key_1": ["obs_1", "obs_2", "obs_3"],
                "data_key_2": ["obs_4"],
                "All_obs": ["obs_1", "obs_2", "obs_3", "obs_4"],
            },
        ),
        (
            {
                "data_key_1": ["obs_1", "obs_2", "obs_3"],
                "data_key_2": ["obs_4"],
                "data_key_3": ["obs_6"],
            },
            {
                "data_key_1": ["obs_1", "obs_2", "obs_3"],
                "data_key_2": ["obs_4"],
                "data_key_3": ["obs_6"],
                "All_obs-data_key_3": ["obs_1", "obs_2", "obs_3", "obs_4"],
                "All_obs-data_key_2": ["obs_1", "obs_2", "obs_3", "obs_6"],
                "All_obs-data_key_1": ["obs_4", "obs_6"],
                "All_obs": ["obs_1", "obs_2", "obs_3", "obs_4", "obs_6"],
            },
        ),
```

Reply:
ok done